server/src/main/java/org/elasticsearch/plugins/SearchPlugin.java

```diff
@@ -128,6 +129,10 @@ default List<QuerySpec<?>> getQueries() {
         return emptyList();
     }
 
+    default List<QueryRewriteInterceptor> getQueryRewriteInterceptors() {
+        return emptyList();
+    }
```

Comment:
we should also add a comment that this is an internal to Elasticsearch API (though its already protected via the module stuff).